Code/network/include/Connection.h

```diff
@@ -55,22 +61,30 @@ class Connection
     Connection& operator=(Connection&& aRhs) noexcept;
     Connection& operator=(const Connection& aRhs) = delete;
 
-    bool ProcessPacket(Buffer* apBuffer);
-    bool ProcessNegociation(Buffer* apBuffer);
-
+    Outcome<uint64_t, Connection::HeaderErrors> ProcessPacket(Buffer::Reader & aReader);

```

Comment:
uint64_t doesn't explain what it contains, you can typedef something with an explicit name and/or comment what it does